### DISCUSSION TITLE
(Dev) Documentation?

### DISCUSSION THREAD

**Discussion by KarimElsayad247** [zen-browser/desktop#1777] [Ideas]
Is there an existing documentation for the project, from the perspective of Zen changes?

If there isn't, I really suggest introducing some documentation. It will encourage more people to contribute.

The thing that prompted me to look was finding the Zen-Components repo, which also exists in the Desktop repo. I didn't know how to make since of that.

I want to start contributing, so I was also thinking of introducing some docs as I went along since I would have to figure out what does what anyway, so I'm looking for suggestions on how to contribute this documentation.

What do you think? @HarryHeres @mauro-balades how would you envision (dev) documentation for zen?

**Comment by nocanoa** [comment]
Hey have u seen https://docs.zen-browser.app ?
https://github.com/zen-browser/docs

**Reply by KarimElsayad247** [reply]
Yes, however, the contribution guides section is more of a "Getting started with contribution" guide, but I'm interested in Zen-specific code documentation.

For example, let's say I picked up a bug about some padding issue in tbe toolbar. *Where do I start?* Where do I find the code for this toolbar? What about vertical tabbar?  I'll have to go through the code the same way I go through any undocumented project, jumping from folder to folder, and file to file, hoping to find the file that looks promising.

If, instead, there was a guide saying "toolbar implementation is in <link>, vertical tabs implementation is in <link> and so on

Or directory-level readme files, so you get a since of what each directory contains, instead of having to guess based on directory names and contents.

I just learned about surfer, the build too used in Zen, but I had to google it before I found out it's in a Zen repository, after which I went through the docs for gluon which explained some of the directories in the root level of the repo.

You get what mean? Lowering the barrier to entry would provide an opportunity for more people to contribute to Zen.

**Reply by KarimElsayad247** [reply]
This is doubly important since I'm sure the vast majority of developers don't have experience contributing to an open source project, let alone a complex project like Firefox.

**Reply by nocanoa** [reply]
yeah, I get what u mean,  we had a chat in discrd about this and yes it should hapen, but since zen is still changing alot it kinda got puched back : /
@IAmJafeth is the big brain for docs, let him know, and since u sayed
> so I was also thinking of introducing some docs as I went along since I would have to figure out what does what anyway

let him know and u could add some stuff to docs.

btw join discord its faster responce xD

**Comment by Cretezy** [comment]
:+1: for this, the [Building Zen Browser](https://docs.zen-browser.app/guides/building) page explains how to build the browser, but not how to make changes or contribute upstream. The [Contributing](https://docs.zen-browser.app/contribute/contributing) page also is missing documentation on how to contribute to the browser itself.
